plugins/flatpak/gs-flatpak.c

```diff
@@ -3468,6 +3537,7 @@ gs_flatpak_init (GsFlatpak *self)
 	self->broken_remotes = g_hash_table_new_full (g_str_hash, g_str_equal,
 						      g_free, NULL);
 	self->store = as_store_new ();
+	self->volume_monitor = g_volume_monitor_get ();
```

Comment:
Not sure if you will end up needing this, but if so it's leaked.

Reply:
You should also disconnect your signal handler when dropping your reference to the `GVolumeMonitor`, in case you don’t hold the last reference to it.